src/configuration/app/source-operations.md

```diff
@@ -0,0 +1,68 @@
+---
+search:
+    keywords: ['.platform.app.yaml', 'automated', 'code', 'source', 'update', 'operation', 'dependencies', 'auto']
+---
+
+# Source operations
+
+> **note**
+>
+> Source Operations are currently in beta.  While the syntax is not expected to change some behavior might in the future.
+
+An application can define a number of operations that apply to its source code and that can be automated.
+
+A basic, common source operation could be to automatically update Composer dependencies like this:
+
+```yaml
+source:
+    operations:
+        update:
+            command: |
+                composer update
+                git add composer.lock
+                git commit -m "Update Composer dependencies."
+```
+
+The `update` key is the name of the operation. It is arbitrary, and multiple source operations can be defined.
+
+The environment resource gets a new `source-operation` action which can be triggered by the CLI:
+
+```
+platform source-operation:run update
+```
+
+The command has two parameters:
```

Comment:
It feels like you're talking about the CLI command there, but then you mention `variables`, so you actually mean the YAML? Maybe you should say "The `command` key"?